### PR TITLE
feat: add fragmentNavigated event

### PR DESCRIPTION
**Pull request by Lightning00Blade**
Unblocks `waitForNavigation` in Puppeteer.

After talking with @sadym-chromium we should only give navigation to the initial request (document request).
But all CDP requests triggered by initial parsing of the document have the LoaderId set, so we relay on requestId matching the loaderId.

### CODE REVIEW

File: src/bidiMapper/domains/network/networkRequest.ts [hunk 1]
```diff
@@ -170,6 +170,20 @@ export class NetworkRequest {
     };
   }
 
+  #getNavigationId(): BrowsingContext.Navigation | null {
+    if (
+      !this.#requestWillBeSentEvent?.loaderId ||
+      // All request that are initiated by the navigation
+      // receive loaderId we want only the initial request
+      // to have the navigation set
```
Comment (thiagowfx) [inline src/bidiMapper/domains/network/networkRequest.ts:178]: Could you rephrase this? It sounds a bit confusing.

**Review by thiagowfx — CHANGES_REQUESTED** [review]
It would be nice to add at least one test that exercises the newly added event

(or augment an existing test)

File: tests/test_browsing_context.py [hunk 2]
```diff
@@ -922,3 +922,37 @@ async def test_browsingContext_ignoreCache(websocket, context_id, ignoreCache):
...
         "result": {},
     }
+
+
+@pytest.mark.asyncio
+async def test_browsingContext_fragmentNavigated_event(websocket, context_id,
+                                                       html):
+    url = html()
+
+    await subscribe(websocket, ["browsingContext.fragmentNavigated"])
```
Comment (thiagowfx) [inline tests/test_browsing_context.py:932]: Nit: when listening to a single element,  you can pass a string directly (instead of an array).

**Review by thiagowfx — APPROVED** [review]

File: tests/test_browsing_context.py (hunk 2 above)
Comment (Lightning00Blade) [reply tests/test_browsing_context.py:932]: ack
